haskell/src/Monocle/Backend/Test.hs

```diff
@@ -547,6 +550,139 @@ testGetSuggestions = withTenant doTest
+                    tdTitle = ""
+                  }
+              ]
+          )
+        ]
+        ((\ELKChange {..} -> (elkchangeId, elkchangeTasksData)) <$> changes)
+      -- Ensure associated ChangeEvents got the Task data attibutes
+      events <- I.getChangesEventsByURL (map ("https://fakeprovider/" <>) ["42", "43", "44"]) 100
+      let (withTD, withoutTD) = partition (isJust . elkchangeeventTasksData) events
+          createdEventWithTD =
+            filter
+              (\e -> (e & elkchangeeventType) == ElkChangeCreatedEvent)
+              withTD
+      assertEqual' "Check events count that got a Task data" 8 (length withTD)
+      assertEqual' "Check events count that miss a Task data" 4 (length withoutTD)
+      assertEqual'
+        "Check Change events got the task data attribute"
+        [ ( "ChangeCreatedEvent-42",
+            Just
+              [ ELKTaskData
+                  { tdTid = "",
+                    tdTtype = [],
+                    tdChangeUrl = "https://fakeprovider/42",
+                    tdSeverity = "",
+                    tdPriority = "",
+                    tdScore = 0,
+                    tdUrl = "https://tdprovider/42-42",
+                    tdTitle = ""
+                  }
+              ]
+          ),
+          ( "ChangeCreatedEvent-43",
+            Just
+              [ ELKTaskData
+                  { tdTid = "",
+                    tdTtype = [],
+                    tdChangeUrl = "https://fakeprovider/43",
+                    tdSeverity = "",
+                    tdPriority = "",
+                    tdScore = 0,
+                    tdUrl = "https://tdprovider/42-43",
+                    tdTitle = ""
+                  }
+              ]
+          )
+        ]
+        ( ( \ELKChangeEvent {..} ->
+              (elkchangeeventId, elkchangeeventTasksData)
+          )
+            <$> createdEventWithTD
+        )
+
+      -- Send a Task data w/o a matching change
+      let td = mkTaskData "45"
+      void $ I.taskDataAdd [td]
+      -- Ensure the Task data has been stored as orphan (we can find it by its url as DocId)
+      orphanTdM <- getOrphanTd . toText $ td & taskDataUrl
+      assertEqual'
+        "Check Task data stored as Orphan Task Data"
+        ( Just
+            ( ELKChangeOrphanTD
+                { elkchangeorphantdType = ElkOrphanTaskData,
+                  elkchangeorphantdTasksData =
+                    Just
+                      [ ELKTaskData
+                          { tdTid = "",
+                            tdTtype = [],
+                            tdChangeUrl = "https://fakeprovider/45",
+                            tdSeverity = "",
+                            tdPriority = "",
+                            tdScore = 0,
+                            tdUrl = "https://tdprovider/42-45",
+                            tdTitle = ""
+                          }
+                      ]
+                }
+            )
+        )
+        orphanTdM
```

Comment:
I guess we are missing the orphaned assignment in the new change add api? perhaps we should leave a `TODO: check adding change 45 gets the td and that it clears the orphan list.`

Reply:
Yes, I think this can be done when we will make the Gerrit and Github crawler to use the Crawler API.  At that time the  Crawler AddDocRequest function will get that capability to look if a input change match an orphan task data then will if this is the case will associate the orphan to the change and set the orphan as adopted.
No need for the TODO imo.